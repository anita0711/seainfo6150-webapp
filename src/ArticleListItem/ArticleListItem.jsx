import React, { useState } from "react";
import PropTypes from "prop-types";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton.jsx";
import styles from "./ArticleListItem.module.css";

const ArticleListItem = (props) => {
  const [text, setText] = useState(() => {
    return "Show more";
  });

  function toggleText() {
    setText((prevText) =>
      prevText === "Show more" ? "Show less" : "Show more"
    );
  }

  let content;
  if (text === "Show less") {
    content = (
      <>
        <p>{props.article.shortText}</p>
        <time dateTime={props.article.timeStamp} className={styles.displayDate}>
          {props.article.displayDate}
        </time>
        <br /> <br />
      </>
    );
  }

  return (
    <div className={styles.container}>
      <h2 className={styles.title}>{props.article.title}</h2>
      {content}
      <ArticleTextToggleButton onClick={toggleText} buttonText={text} />
    </div>
  );
};

ArticleListItem.propTypes = {
  article: PropTypes.object.isRequired,
};
export default ArticleListItem;
