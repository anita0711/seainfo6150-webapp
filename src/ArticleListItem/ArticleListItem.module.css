.container {
  border: 1px solid #c00;
  margin-bottom: 20px;
  padding: 10px;
  position: relative;
  border-radius: 15px;
  background-image: linear-gradient(to right, rgb(245, 91, 91), #eee 15%);
  font-family: Arial, Helvetica, sans-serif;
}

.title {
  color: #c00;
  font-weight: bold;
  margin: 0.5rem 0;
}

.displayDate {
  font-weight: bold;
}
