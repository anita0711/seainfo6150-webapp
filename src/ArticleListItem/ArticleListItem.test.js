import React from "react";
import ArticleListItem from "./ArticleListItem";

const article = {
  author: "Roxie Parks",
  displayDate: "November 21st 2018, 6:43 pm",
  shortText:
    "On a dusty, 3,600-foot-long airstrip in Rajasthan, India, a two-seater white Cessna 152 pivots into position. A few moments later, it's soaring over the empty desert into a cloudless blue sky.",
  title: "Training India's next generation of female pilots",
};

describe("ArticleListItem tests", () => {
  it("renders correctly after object is passed", () => {
    const { container } = render(<ArticleListItem article={article} />);
    expect(container).toMatchSnapshot();
  });
});
