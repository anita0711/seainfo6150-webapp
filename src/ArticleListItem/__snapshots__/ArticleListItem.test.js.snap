// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly after object is passed 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <h2
          class="title"
        >
          Training India's next generation of female pilots
        </h2>
      </div>
      <button
        class="button"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
